components/encryption/src/master_key/mem.rs

```diff
@@ -53,7 +53,7 @@ impl MemAesGcmBackend {
                 ))
             })?;
         if method.as_slice() != MetadataMethod::Aes256Gcm.as_slice() {
-            return Err(Error::WrongMasterKey(box_err!(
+            return Err(Error::Other(box_err!(
```

Comment:
I'm confused, is there any rule to determining if we should return a `WrongMasterKey` error?

Reply:
The rule in my mind is return WrongMasterKey whenever the error is possibly caused by a wrong master key used, and other error otherwise. Here since only aes256-gcm is allowed currently, if we encounter a new method, the encrypted content could be generated by a future version of TiKV, where we cannot handle anyway even we fallback to a different master key. I added some comments to reason able the error handling.